core/blockchain.go

```diff
@@ -591,15 +593,17 @@ func (bc *BlockChain) SetHead(head uint64) error {
 }
 
 func (bc *BlockChain) tryRewindBadBlocks() {
-	bc.chainmu.Lock()
-	defer bc.chainmu.Unlock()
```

Comment:
lock scope has been changed. but it should be ok.

Reply:
yes, a bit of refactor on purpose